UnityProject/Assets/Scripts/HealthV2/Living/BodyParts/RootBodyPartContainer.cs

```diff
@@ -339,16 +352,16 @@ public virtual void SubBodyPartAdded(BodyPart implant)
 			//This is so you can still hit for example the Second Head of a double-headed thing, can be changed if we find a better solution for aiming at Specific body parts
 			if (damageSplit || attackType == AttackType.Bomb || attackType == AttackType.Fire || attackType == AttackType.Rad)
 			{
-				foreach (var ContainsLimb in ContainsLimbs)
+				//We don't use foreach to avoid errors when the list gets modifed.
+				for(int limbCount = 0; ContainsLimbs.Count > limbCount; limbCount++)
```

Comment:
Some items in the list may be skipped under the current implementation. 

If an item is removed from the list, `limbCount` already points to the next item in the list. But `limbCount` is also increased at the end of the loop. So we skip the item in the list.

The simplest solution is to go through the list from the end, for example, like this: `for (int limbCount = ContainsLimbs.Count - 1; limbsCount >= 0; limbCount--)`.

By the way, it is better to also make this list a LinkedList. Removing items from a middle of a regular list can make the garbage collector suffer. In this case, it is better to use your `foreach ()` implementation via `Enumerator`

Reply:
I've added your solution of looping the thing backwards, LinkedList doesn't seem to be working as intended so I'm going to leave it as that for now.

Reply:
> foreach () implementation via Enumerator

How would that look @Lizurt ?

Reply:
> How would that look @Lizurt ?

I worked with `Enumerator` (Iterator) only in Java, so it looks like we can't use the `Enumerator` here, since it goes through the values, not the nodes. You will have to go down even one more level and use your own abstract `Enumerator`. Something like(a bit pseudocoded):
```
LinkedListNode<Limb> limbNode = list.First;
LinkedListNode<Limb> nextLimbNode;
while (limbNode != null)
{
    nextNode = node.Next;
    limbNode.Value.TakeDamage(args);
    node = nextNode;
}
```

You will also have to delete a limb in the list by node and not by its value, unless you want the complexity O(n) instead of O(1) when removing.

Reply:
Remember that this is just a recommendation. Since the list is small and deletions are rare, it's not so critical to leave it as a simple list imo.

Reply:
I have tested it and it works from a gameplay perspective, cant speak towards the looping method.